TombEngine/Objects/TR5/Emitter/tr5_smoke_emitter.cpp

```diff
@@ -83,14 +83,15 @@ void SmokeEmitterControl(short itemNumber)
 		{
 			if (!(GetRandomControl() & 3) || item->ItemFlags[1])
 			{
-				pos.x = (GetRandomControl() & 0x3F) + item->Pose.Position.x - 32;
-				pos.y = item->Pose.Position.y - (GetRandomControl() & 0x1F) - 16;
-				pos.z = (GetRandomControl() & 0x3F) + item->Pose.Position.z - 32;
+				auto pos = Vector3(
+					(GetRandomControl() & 0x3F) + item->Pose.Position.x - 32,
+					item->Pose.Position.y - (GetRandomControl() & 0x1F) - 16,
+					(GetRandomControl() & 0x3F) + item->Pose.Position.z - 32);
 
 				if (item->TriggerFlags == 1)
-					CreateBubble(&pos, item->RoomNumber, 15, 15, 0, 0, 0, 0);
+					SpawnBubble(pos, item->RoomNumber, 0);
 				else
-					CreateBubble(&pos, item->RoomNumber, 8, 7, 0, 0, 0, 0);
+					SpawnBubble(pos, item->RoomNumber, 0);
```

Comment:
I'm misunderstanding these.

So Smoke emitter when placed underwater can have OCB 1 or any other (OCB 0 for example). What should be the difference?

In the old function, I see it's passing different parameters as arguments, but I'm not sure how the new SpawnBubble function is recreating one or the second. 

(And if it happens that they are really the same, couldn't ignore the OCB and leave just 1 call?).

Reply:
Later in the test I checked.

I used SMOKE_EMITTER_WHITE but I saw it was using also the same Control function.

- Smoke_Emitter_White OCB 1, creates a continium bubbles emission.
- Smoke_Emitter_White OCB 0, emits the bubbles in pulses.

Reply:
Hmm, I just noticed these are the same now. The removed arguments were both unused and unknown in the original decompiled version. I'll check tomb4/5 to see if they're supposed to do anything.

In your later test, is that behaviour observed with this code or with the previous?

Reply:
The extra arguments don't seem to be anything important, so I'll leave the logic here as is.